ray-operator/main.go

```diff
@@ -117,15 +119,29 @@ func main() {
 		setupLog.Info("Feature flag enable-batch-scheduler is enabled.")
 	}
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	watchNamespaces := strings.Split(watchNamespace, ",")
+	options := ctrl.Options{
 		Scheme:                 scheme,
 		MetricsBindAddress:     metricsAddr,
 		Port:                   9443,
 		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
 		LeaderElectionID:       "ray-operator-leader",
-		Namespace:              watchNamespace,
-	})
+	}
+
+	if len(watchNamespaces) == 1 {
+		options.Namespace = watchNamespaces[0]
+		if watchNamespaces[0] == "" {
+			setupLog.Info("Flag watchNamespace is not set. Watch custom resources in all namespaces.")
+		} else {
+			setupLog.Info(fmt.Sprintf("Only watch custom resources in the namespace: %s", watchNamespaces[0]))
+		}
+	} else {
+		options.NewCache = cache.MultiNamespacedCacheBuilder(watchNamespaces)
```

Comment:
To watch multiple namespaces, we can use MultiNamespacedCache. See [this example](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/manager#example-New-MultinamespaceCache) and [this doc](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/cache?utm_source=godoc#MultiNamespacedCacheBuilder) for more details.